Tools/TestWebKitAPI/Tests/ios/ScrollViewBouncesTests.mm

```diff
@@ -0,0 +1,191 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+
+#if PLATFORM(IOS_FAMILY)
+
+#import "PlatformUtilities.h"
+#import "TestInputDelegate.h"
+#import "TestNavigationDelegate.h"
+#import "TestWKWebView.h"
+#import <UIKit/UIKit.h>
+#import <WebKit/WKWebViewPrivate.h>
+
+namespace TestWebKitAPI {
+
+static const CGFloat viewHeight = 500;
+
+static NSString *overscrollBehaviorAuto = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior: auto;'>";
+static NSString *overscrollBehaviorNone = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior: none;'>";
+static NSString *overscrollBehaviorContain = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior: contain;'>";
+
+static NSString *overscrollBehaviorAutoX = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-x: auto;'>";
+static NSString *overscrollBehaviorNoneX = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-x: none;'>";
+static NSString *overscrollBehaviorContainX = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-x: contain;'>";
+
+static NSString *overscrollBehaviorAutoY = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-y: auto;'>";
+static NSString *overscrollBehaviorNoneY = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-y: none;'>";
+static NSString *overscrollBehaviorContainY = @"<meta name='viewport' content='width=device-width, initial-scale=1'><body style='width: 100%; height: 5000px; overscroll-behavior-y: contain;'>";
+
+TEST(ScrollViewBouncesTests, OverscrollBehaviorAutoNotSet)
+{
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, viewHeight)]);
+    [webView synchronouslyLoadHTMLString:overscrollBehaviorAuto];
+    EXPECT_EQ([[webView scrollView] bounces], YES);
+}
+
+TEST(ScrollViewBouncesTests, OverscrollBehaviorAutoSet)
+{
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, viewHeight)]);
+    [webView synchronouslyLoadHTMLString:overscrollBehaviorAuto];
+    [[webView scrollView] setBounces:NO];
```

Comment:
How about a test that calls -setBounces: BEFORE the load?

You could also test that a page that dynamically changes `overscroll-behavior` does the right thing.

Reply:
Fixed.